src/realm/object-store/schema.cpp

```diff
@@ -279,10 +284,16 @@ static void compare(ObjectSchema const& existing_schema, ObjectSchema const& tar
     }
 }
 
-template <typename T, typename U, typename Func>
-void Schema::zip_matching(T&& a, U&& b, Func&& func) noexcept
+template <typename Func>
+std::vector<ObjectSchema> Schema::zip_matching(Schema const& a, Schema const& b, Func&& func, bool is_schema_additive)
```

Comment:
I would not change the functionality of this function, but rather handle the added Schemas in the callback function.

Reply:
this has been resolved.